Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +9,234 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            var splitArray = peopleAndBalances[0].Split(',');
+            var floatArray = ConvertStringArrayToFloat(splitArray[1..]);
+            var names = new[] { splitArray[0] };
+            var heighest = FindHighest(floatArray);
+
+            for (int i = 1; i < peopleAndBalances.Length; i++)
+            {
+                var currrentSplitArray = peopleAndBalances[i].Split(',');
+                var currentFloatArray = ConvertStringArrayToFloat(currrentSplitArray[1..]);
+                var name = currrentSplitArray[0];
+                var currentHeighest = FindHighest(currentFloatArray);
+
+                if (currentHeighest > heighest)
+                {
+                    heighest = currentHeighest;
+                    names = new[] { name };
+                }
+                else if (currentHeighest == heighest)
+                {
+                    names = AddStrings(names, name);
+                }
+            }
+
+            return $"{FormatStrings(names)} had the most money ever. ¤{heighest}.";
+        }
+
+        private static float FindHighest(float[] floatArray)
+        {
+            float currentMax = floatArray[0];
+            for (int i = 1; i < floatArray.Length; i++)
+            {
+                if (floatArray[i] > currentMax)
+                {
+                    currentMax = floatArray[i];
+                }
+            }
+            return currentMax;
+        }
+
+        private static float[] ConvertStringArrayToFloat(string[] v)
+        {
+            var arr = new float[v.Length];
+
+            for (int i = 0; i < v.Length; i++)
+            {
+                arr[i] = float.TryParse(v[i], out float parsed) ? parsed : 0f;
+            }
+            return arr;
+        }
+
+        private static string FormatStrings(string[] currentNames)
+        {
+            if (currentNames.Length == 1)
+            {
+                return currentNames[0];
+            }
+            var formatArray = String.Join(", ", currentNames[..^1]);
+            formatArray += $" and {currentNames[^1]}";
+
+            return formatArray;
+        }
+
+        private static string[] AddStrings(string[] currentNames, string v)
```

Comment:
A method such as this would expect 2 strings for parameters. However, this has array of string and a string. I recommend calling this AddString instead, because you add just one string.

Reply:
Good point, changed.